src/Tribe/RSVP.php

```diff
@@ -580,20 +909,30 @@ public function generate_tickets() {
 
 		// If there are details missing, we return to the event page with the rsvp_error
 		if ( false === $attendee_details ) {
-			$url = get_permalink();
-			$url = add_query_arg( 'rsvp_error', 1, $url );
-			wp_redirect( esc_url_raw( $url ) );
-			tribe_exit();
+			if ( $redirect ) {
+				$url = get_permalink();
+				$url = add_query_arg( 'rsvp_error', 1, $url );
+				wp_redirect( esc_url_raw( $url ) );
+				tribe_exit();
+			}
+
+			return;
 		}
 
+		$product_ids = (array) $_POST['product_id'];
```

Comment:
* :warning: Detected usage of a non-sanitized input variable: $_POST['product_id']
* :no_entry_sign: Processing form data without nonce verification.
* :no_entry_sign: Detected usage of a possibly undefined superglobal array index: $_POST['product_id']. Use isset() or empty() to check the index exists before using it

<sub>hash: 20e0f58c54a3be28457f740da56d1890</sub>